analyzer/src/main/kotlin/curation/OrtConfigPackageCurationProvider.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2022 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.analyzer.curation
+
+import com.fasterxml.jackson.module.kotlin.readValue
+
+import java.io.File
+
+import org.ossreviewtoolkit.analyzer.PackageCurationProvider
+import org.ossreviewtoolkit.downloader.vcs.Git
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.PackageCuration
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.VcsType
+import org.ossreviewtoolkit.model.yamlMapper
+import org.ossreviewtoolkit.utils.common.encodeOr
+import org.ossreviewtoolkit.utils.common.safeMkdirs
+import org.ossreviewtoolkit.utils.core.ortDataDirectory
+
+private const val ORT_CONFIG_REPOSITORY_BRANCH = "main"
+private const val ORT_CONFIG_REPOSITORY_URL = "https://github.com/oss-review-toolkit/ort-config.git"
+
+/**
+ * A [PackageCurationProvider] that provides [PackageCuration]s loaded from the
+ * [ort-config repository](https://github.com/oss-review-toolkit/ort-config).
+ */
+open class OrtConfigPackageCurationProvider : PackageCurationProvider {
+    private val curationsDir by lazy {
+        ortDataDirectory.resolve("ort-config").also {
+            updateOrtConfig(it)
+        }
+    }
+
+    override fun getCurationsFor(pkgIds: Collection<Identifier>) =
+        pkgIds.mapNotNull { pkgId ->
+            getCurationsFor(pkgId).takeIf { it.isNotEmpty() }?.let { pkgId to it }
+        }.toMap()
+
+    private fun getCurationsFor(pkgId: Identifier): List<PackageCuration> {
+        val file = curationsDir.resolve("curations").resolve(pkgId.toCurationPath())
+        return if (file.exists()) {
+            yamlMapper.readValue(file)
+        } else {
+            emptyList()
+        }
+    }
+}
+
+private fun Identifier.toCurationPath() =
+    "${type.encodeOr("_")}/${namespace.encodeOr("_")}/${name.encodeOr("_")}.yml"
```

Comment:
Has there already been a decision made to use this file structure for that repository?

In order to fail early I believe it'd be good to load all the curations upfront and keep them in memory with a `map`. Therefore `SimplePackageConfigurationProvider.forDirectory()` could be re-used if slighly refactored. 

This would avoid enforcing a directory structure, so that all files under a given directory are considered.

Reply:
I wonder if you can drop this class in favor of a static function in SimplePackageCurationProvider, e.g. 
`forOrtConfigRespository()`

Reply:
Yes, the repository enforces a certain structure which makes sense for a public repository. The exact structure is subject to further discussion of course, but for now it is that. Given that, it is just unnecessary to load all curations into memory, especially since hopefully a LOT of curations will be added to that repository.

Reply:
> Yes, the repository enforces a certain structure which makes sense for a public repository. 

It's out of question that it makes sense to enforce a certain structure in the repository. That was not my point. I was rather after whether it makes sense to make this ORT code reflect all these details or whether it should be kept a bit more generic.

> Given that, it is just unnecessary to load all curations into memory, especially since hopefully a LOT of curations will be added to that repository.

As you say "unnecessary" this implies to me you mean that it does not provide any value. From my comments above I was arguing that it does provide value, in particular it (1) implements fail early and (2) allows flexibility to the file structure
which might also make sense with changing it because: 

> The exact structure is subject to further discussion of course,

Reply:
That's right, but I don't expect that the structure changes regularly given the effort required to migrate all files, I would more expect that it changes maybe once or twice at all, so it wouldn't be hard to keep in sync. The fail early part can also be bad, if it fails because of curations which are not required for your project at all. However, as the config repo verifies that curation files can be parsed it should actually never fail, except if there are breaking changes in the file format, but I think we would always want to implement those in a backward compatible way.

Reply:
Btw, I don't consider this to be the final solution for integrating the ort-config repo, it's just a quick implementation to be able to start using it. Once more files types like package configurations or reporter templates are added we might have to think about a more sophisticated integration anyway.

Reply:
I still tend to rather consider assuming a file organization is "unnecessary" compared to parsing all `yaml` files within the curations directory, in particular I don't see any performance or memory consumption issues with holding these in memory. Anyhow, let's just disagree on that one and continue.